core/src/main/java/org/infinispan/container/entries/ClusteredRepeatableReadEntry.java

```diff
@@ -71,6 +71,25 @@ public boolean performWriteSkewCheck(DataContainer container, PersistenceManager
       return InequalVersionComparisonResult.EQUAL == result;
    }
 
+   private EntryVersion getCurrentEntryVersion(DataContainer container, PersistenceManager persistenceManager, TxInvocationContext ctx, VersionGenerator versionGenerator, TimeService timeService) {
+      EntryVersion prevVersion;// on origin, the version seen is acquired without the lock, so we have to retrieve it again
+      // TODO: persistence should be more orthogonal to any entry type - this should be handled in interceptor
```

Comment:
I agree `CacheEntry` implementations in general shouldn't know about the persistence, but after a while I've come to see `MVCCEntry` implementations more as part of `EntryWrappingInterceptor` :)